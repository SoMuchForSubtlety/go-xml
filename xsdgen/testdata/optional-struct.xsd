<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ns="http://example.org"
  targetNamespace="http://example.org">


  <xs:element name="TypeB">
    <xs:complexType>
      <xs:sequence>
        <xs:element minOccurs="0" nillable="true" name="deliveryDays" type="ns:DateRange">
          <xs:annotation>
            <xs:documentation>Range of Delivery Days the Order applies to.</xs:documentation>
          </xs:annotation>
        </xs:element>
        <xs:element maxOccurs="unbounded" minOccurs="0" name="errors" nillable="true"
          type="ns:Error">
          <xs:annotation>
            <xs:documentation>Range of Delivery Days the Order applies to.</xs:documentation>
          </xs:annotation>
        </xs:element>
        <xs:element name="xxx" type="ns:ooo" minOccurs="0" maxOccurs="unbounded"></xs:element>
        <xs:element name="stringField">
          <xs:simpleType>
            <xs:restriction base="xs:string">
            </xs:restriction>
          </xs:simpleType>
        </xs:element>
      </xs:sequence>
    </xs:complexType>
  </xs:element>

  <xs:complexType name="Error">
    <xs:sequence>
      <xs:element name="text" type="xs:string">
      </xs:element>
      <xs:element name="code" type="xs:string">
      </xs:element>
    </xs:sequence>
  </xs:complexType>

  <xs:simpleType name="ooo">
    <xs:restriction base="xs:string" />
  </xs:simpleType>


  <xs:complexType name="DateRange">
    <xs:annotation>
      <xs:documentation>A Range Object that indicates a Date Range the date Struct has to and from
        as Date Types</xs:documentation>
    </xs:annotation>
    <xs:sequence>
      <xs:element name="from" type="xs:string">
        <xs:annotation>
          <xs:documentation>Starting date</xs:documentation>
        </xs:annotation>
      </xs:element>
      <xs:element name="to" type="xs:string">
        <xs:annotation>
          <xs:documentation>Ending date</xs:documentation>
        </xs:annotation>
      </xs:element>
    </xs:sequence>
  </xs:complexType>
</xs:schema>
