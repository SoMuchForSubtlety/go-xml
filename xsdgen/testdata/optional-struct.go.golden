// Code generated by xsdgen.test. DO NOT EDIT.

package ws

// A Range Object that indicates a Date Range the date Struct has to and from
// as Date Types
type DateRange struct {
	From string `xml:"http://example.org from"`
	To   string `xml:"http://example.org to"`
}

type Error struct {
	Text string `xml:"http://example.org text"`
	Code string `xml:"http://example.org code"`
}

type TypeB struct {
	DeliveryDays *DateRange `xml:"http://example.org deliveryDays,omitempty"`
	Errors       []Error    `xml:"http://example.org errors,omitempty"`
	Xxx          []string   `xml:"http://example.org xxx,omitempty"`
	StringField  string     `xml:"http://example.org stringField"`
}
